### PR TITLE
[Raft] Documentation updates related to the Pre-vote stage implementation 

### LINKED ISSUE #2593: [1sp] Raft: document the new Pre-Vote stage in leader election process
**Epic:** https://github.com/tarantool/doc/issues/2592

**Product:** Tarantool
**Since:** 2.10.0
**Dev issue:** https://github.com/tarantool/tarantool/issues/6654
**Root document:** https://www.tarantool.io/en/doc/latest/book/replication/repl_leader_elect/#leader-election-process
**SME:** @ sergepetrenko 

# Details

[input from dev issue]
When a server gets partitioned from the majority of the cluster, it starts incrementing its term every `election_timeout`, trying to win elections. Once such a server reunites with the cluster it unintentionally disrupts the current working leader. The leader steps off seeing a greater term number. This makes the whole cluster go through at least one round of elections, rendering it read-only for a couple of seconds.

Diego Ongaro's [thesis](https://github.com/ongardie/dissertation#readme) covers this issue and suggests adding a Pre-Vote stage to Raft

**TDB**  details of the implementation of the Pre-Vote stage in Tarantool

# ToDo

- Update the description of the [leader election process](https://www.tarantool.io/en/doc/latest/book/replication/repl_leader_elect/#leader-election-process) starting from this sentence: "So if there are no heartbeats for a period set by the replication_timeout option, a new election starts." The process will be different with the Pre-Vote stage

### PR DESCRIPTION
**Pull request by veod32**

### CODE REVIEW

**Comment by veod32** [comment]
Changes reported in #2837 are also part of this PR.

File: doc/book/replication/repl_leader_elect.rst [hunk 1]
```diff
@@ -69,10 +69,15 @@ Eventually, a leader is elected.
 
 All the non-leader nodes are called *followers*. The nodes that start a new
 election round are called *candidates*. The elected leader sends heartbeats to
-the non-leader nodes to let them know it is alive. So if there are no heartbeats
-for a period set by the :ref:`replication_timeout <cfg_replication-replication_timeout>`
-option, a new election starts. Terms and votes are persisted by
-each instance in order to preserve certain Raft guarantees.
+the non-leader nodes to let them know it is alive.
+
+In case there are no heartbeats for a period set by the :ref:`replication_timeout <cfg_replication-replication_timeout>` option,
```
Comment (sergepetrenko) [inline doc/book/replication/repl_leader_elect.rst]: An election is started if there are no heartbeat for `4 * replication_timeout`. This value is denoted `replication_death_timeout` internally, but it is not a configuration option, and I don't know if it's mentioned anywhere in the doc or not.

File: doc/book/replication/repl_leader_elect.rst [hunk 2]
```diff
@@ -69,10 +69,15 @@ Eventually, a leader is elected.
...
+a non-leader node starts a new election if the following conditions are met:
+
+*   It has a quorum of connections to other cluster members.
+*   None of these cluster members can see the leader node.
```
Comment (sergepetrenko) [inline doc/book/replication/repl_leader_elect.rst:81]: I don't know if it's worth mentioning or not. You decide:
The other cluster members think that they see the leader node, if they have heard from it at least once during the `replication_death_timeout` period AND there were no replication errors (the connection is not broken due to timeout or due to an error).

File: doc/reference/reference_lua/box_info/election.rst [hunk 3]
```diff
@@ -10,9 +10,25 @@ box.info.election
...
+
+   The following information is provided:
+
+   *    ``state`` -- election state (mode) of the node. Possible values are ``leader``, ``follower``, or ``candidate``.
+        For more details, refer to description of the :ref:`leader election process <repl_leader_elect_process>`.
+        When election is enabled, the node is writable only in the ``leader`` state.
+
+   *    ``term`` -- current election term.
+
+   *    ``vote`` -- ID of a node the current node votes for. If the value is ``0``, it means the node doesn't vote in the current term.
```
Comment (sergepetrenko) [inline doc/reference/reference_lua/box_info/election.rst]: If the value is `0`, it means the node hasn't yet voted in the current term.

**Review by sergepetrenko — CHANGES_REQUESTED** [review]

File: doc/reference/reference_lua/box_info/election.rst (hunk 3 above)
Comment (veod32) [reply doc/reference/reference_lua/box_info/election.rst]: Corrected.

File: doc/book/replication/repl_leader_elect.rst (hunk 1 above)
Comment (veod32) [reply doc/book/replication/repl_leader_elect.rst]: Corrected.

File: doc/book/replication/repl_leader_elect.rst (hunk 2 above)
Comment (veod32) [reply doc/book/replication/repl_leader_elect.rst:81]: Let's mention these conditions in a note. Please check it out.

**Review by sergepetrenko — APPROVED** [review]
Thanks!

**Pull request merged**